data/src/test/java/org/apache/iceberg/TestMergingMetrics.java

```diff
@@ -110,22 +110,17 @@ public abstract class TestMergingMetrics<T> {
           MAP_FIELD_2,
           STRUCT_FIELD);
 
-  protected final FileFormat fileFormat;
-
-  @Parameterized.Parameters(name = "fileFormat = {0}")
-  public static Object[] parameters() {
-    return new Object[] {FileFormat.PARQUET, FileFormat.ORC};
-  }
+  protected abstract FileAppender<T> writeAndGetAppender(List<Record> records) throws Exception;
 
-  public TestMergingMetrics(FileFormat fileFormat) {
-    this.fileFormat = fileFormat;
+  @Parameters(name = "fileFormat = {0}")
+  public static Collection<FileFormat> parameters() {
```

Comment:
I don't think there's a particular reason to change the signature of this method. Just updating it to the correct `@Parameters` should be enough here

Reply:
It looks to me that most places in the code that have already been converted to Junit 5 use the signature
```
@Parameters(...)
private static List<Object> parameters() {...}
```

The old signature with `Object[]` doesn't seem to work with Junit 5.

I updated all my changes to return `List<Object>` to match the rest of the code. Pls LMK if that's OK.